test/api/clustering_methods_test.cpp

```diff
@@ -0,0 +1,336 @@
+#include <gtest/gtest.h>
+
+#include <seqan3/core/debug_stream.hpp>
+
+#include "modules/clustering/hierarchical_clustering_method.hpp"
+
+using seqan3::operator""_dna5;
+
+
+const std::string chrom1 = "chr1";
+const int32_t chrom1_position1 = 12323443;
+const int32_t chrom1_position2 = 94734377;
+const int32_t chrom1_position3 = 112323345;
+const std::string chrom2 = "chr2";
+const int32_t chrom2_position1 = 234432;
+const std::string & read_name_1 = "m2257/8161/CCS";
+const std::string & read_name_2 = "m41327/11677/CCS";
+const std::string & read_name_3 = "m21263/13017/CCS";
+const std::string & read_name_4 = "m38637/7161/CCS";
+const std::string & read_name_5 = "m23412/9534/CCS";
+const std::string & read_name_6 = "m1245/5634/CCS";
+const std::string & read_name_7 = "m8765/9765/CCS";
+const std::string & read_name_8 = "m13456/11102/CCS";
```

Comment:
The references are pointing to temporary objects, and I believe are invalid after startup.
```suggestion
const std::string read_name_1 = "m2257/8161/CCS";
const std::string read_name_2 = "m41327/11677/CCS";
const std::string read_name_3 = "m21263/13017/CCS";
const std::string read_name_4 = "m38637/7161/CCS";
const std::string read_name_5 = "m23412/9534/CCS";
const std::string read_name_6 = "m1245/5634/CCS";
const std::string read_name_7 = "m8765/9765/CCS";
const std::string read_name_8 = "m13456/11102/CCS";
```

Reply:
Well, the references should extend the lifetime of the temporary objects so that they are accessible as long as the reference lives. But it would be simpler to just use values as you suggested.

Reply:
Yes, you are completely right! @Irallia I claimed something else in different PR, but that was wrong. 
It seems that a "const reference" extends the lifetime
https://herbsutter.com/2008/01/01/gotw-88-a-candidate-for-the-most-important-const/

Reply:
It's also in the C++ reference: https://en.cppreference.com/w/cpp/language/reference

*Rvalue references can be used to extend the lifetimes of temporary objects (note, lvalue references to const can extend the lifetimes of temporary objects too, but they are not modifiable through them)*